qtservice/src/qtservice_unix.cpp

```diff
@@ -58,6 +58,43 @@
 #include <QMap>
 #include <QSettings>
 #include <QProcess>
+#include <QFile>
+
+static bool isServiceInited = false;
+
+static bool isSystemD()
+{
+    return QFileInfo::exists(systemDPath());
+}
+
+static QString systemDPath()
+{
+    return "/etc/systemd/system/";
+}
+
+static void initService()
+{
+
+    if (isSystemD() && !isServiceInited)
+    {
+        QSettings::setPath(QSettings::NativeFormat,
```

Comment:
There are not setPah. It is need caller set. Please see help of QSettings::setPath.
1. The caller is call QSettings::setPath
2. Call QSettings::setPath in Constructor